app/src/main/java/com/epfl/drawyourpath/userProfile/UserModel.kt

```diff
@@ -0,0 +1,354 @@
+package com.epfl.drawyourpath.userProfile
+
+import android.graphics.Color
+import android.text.TextUtils
+import com.epfl.drawyourpath.database.Database
+import java.time.LocalDate
+import java.util.concurrent.TimeUnit
+
+class UserModel {
+    //the userId is given by the authentication part and is unique to the user
+    private val userId: String
```

Comment:
I think it's not a good idea to store the `uid` here. The `auth` object ensures a fresh `uid` that handle disconnecting or account change whenever we access it. I would suggest to simply forward the `uid` from the `Auth` instance.

Reply:
Thank you for the advice, it's modified